### DISCUSSION TITLE
containerSchema missing properties

### DISCUSSION THREAD

**Discussion by pmorskate** [microsoft/FluidFramework#10339] [Q&A]
Ok, so I wanted to mess around with the fluid relay, so I downloaded the angular tutorial repo and did the steps as outlined on the page, but it's not working, even though I have the exact same code as the tutorial suggests.

`Argument of type 'import("C:/Testprojecten/fluid-angular-tutorial/node_modules/fluid-framework/node_modules/@fluidframework/fluid-static/dist/types").ContainerSchema' is not assignable to parameter of type 'import("C:/Testprojecten/fluid-angular-tutorial/node_modules/@fluidframework/fluid-static/dist/types").ContainerSchema'.
  Types of property 'initialObjects' are incompatible.

    Type 'import("C:/Testprojecten/fluid-angular-tutorial/node_modules/fluid-framework/node_modules/@fluidframework/fluid-static/dist/types").LoadableObjectClassRecord' is not assignable to type 'import("C:/Testprojecten/fluid-angular-tutorial/node_modules/@fluidframework/fluid-static/dist/types").LoadableObjectClassRecord'.
      'string' index signatures are incompatible.

        Type 'import("C:/Testprojecten/fluid-angular-tutorial/node_modules/fluid-framework/node_modules/@fluidframework/fluid-static/dist/types").LoadableObjectClass<any>' is not assignable to type 'import("C:/Testprojecten/fluid-angular-tutorial/node_modules/@fluidframework/fluid-static/dist/types").LoadableObjectClass<any>'.

          Type 'DataObjectClass<any>' is not assignable to type 'LoadableObjectClass<any>'.
            Type 'import("C:/Testprojecten/fluid-angular-tutorial/node_modules/fluid-framework/node_modules/@fluidframework/fluid-static/dist/types").DataObjectClass<any>' is not assignable to type 'import("C:/Testprojecten/fluid-angular-tutorial/node_modules/@fluidframework/fluid-static/dist/types").DataObjectClass<any>'.

              Type 'DataObjectClass<any>' is not assignable to type '{ readonly factory: IFluidDataStoreFactory; }'.
                The types of 'factory.instantiateDataStore' are incompatible between these types.
                  Type '(context: import("C:/Testprojecten/fluid-angular-tutorial/node_modules/fluid-framework/node_modules/@fluidframework/runtime-definitions/dist/dataStoreContext").IFluidDataStoreContext, existing: boolean) => Promise<...>' is not assignable to type '(context: import("C:/Testprojecten/fluid-angular-tutorial/node_modules/@fluidframework/runtime-definitions/dist/dataStoreContext").IFluidDataStoreContext, existing: boolean) => Promise<...>'.

                    Types of parameters 'context' and 'context' are incompatible.
                      Property 'raiseContainerWarning' is missing in type 'import("C:/Testprojecten/fluid-angular-tutorial/node_modules/@fluidframework/runtime-definitions/dist/dataStoreContext").IFluidDataStoreContext' but required in type 'import("C:/Testprojecten/fluid-angular-tutorial/node_modules/fluid-framework/node_modules/@fluidframework/runtime-definitions/dist/dataStoreContext").IFluidDataStoreContext'.ts(2345)`

This is the error that is shown around the containerSchema variable in app.component
`    if (!containerId) {
      ({ container } = await client.createContainer(containerSchema));
      const id = await container.attach();
      location.hash = id;
    }`

For good measure, the declaration of containerSchema, which has no errors:
`    const containerSchema: ContainerSchema = {
      initialObjects: { sharedTimestamp: SharedMap }
    };`

Why am I getting this error? The object seems to conform with the ContainerSchema type, right? I've checked, and it is literally a copy of the code on the github.

Thanks for reading!

**Comment by pmorskate** [comment]
Ok, so this seems to be a bit odd, but as soon as I did an explicit import it worked.
`import { ContainerSchema } from "@fluidframework/fluid-static";`

Ok, this didn't fix it, and I'm still trying to figure out what I've done wrong

**Comment by tylerbutler** [comment]
My hunch is it is related to some mismatched dependency versions. Can you share the deps/devdeps for your project?

Also @scottn12 wrote the tutorial originally; maybe he knows something about this.

**Reply by tylerbutler** [reply]
To clarify, I don't think you've done anything wrong. Our package version dependencies are a little fragile at the moment and the tutorial is likely wrong or missing a step.

**Comment by pmorskate** [comment]
Hey Tyler, thanks for the response!

Here's my package.json for the project. I was thinking it might be a version thing, glad to see it wasn't something obvious I've missed.

`{
  "name": "fluid-angular-tutorial",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "watch": "ng build --watch --configuration development",
    "test": "ng test"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "~13.3.0",
    "@angular/common": "~13.3.0",
    "@angular/compiler": "~13.3.0",
    "@angular/core": "~13.3.0",
    "@angular/forms": "~13.3.0",
    "@angular/platform-browser": "~13.3.0",
    "@angular/platform-browser-dynamic": "~13.3.0",
    "@angular/router": "~13.3.0",
    "@fluidframework/tinylicious-client": "^0.47.0",
    "fluid-framework": "^0.47.0",
    "rxjs": "~7.5.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.11.4"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~13.3.6",
    "@angular/cli": "~13.3.6",
    "@angular/compiler-cli": "~13.3.0",
    "@types/jasmine": "~3.10.0",
    "@types/node": "^12.11.1",
    "jasmine-core": "~4.0.0",
    "karma": "~6.3.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage": "~2.1.0",
    "karma-jasmine": "~4.0.0",
    "karma-jasmine-html-reporter": "~1.7.0",
    "typescript": "~4.6.2"
  }
}`

**Reply by scottn12** [reply]
Hi @pmorskate, could you please give your node version? You can get it by running `node -v` in your terminal. I was unable to reproduce this issue on my local with a fresh install and I'm wondering if it has something to do with the node version.

**Reply by pmorskate** [reply]
I am on node 18.2.0 on a fresh install. Recently switched computers

**Reply by scottn12** [reply]
@tylerbutler, can you confirm if we support node 18? Or is it possible that our current release version does but 0.47 does not?

**Reply by mmsharepoint** [reply]
I was able to solve it keeping "fluid-framework", "@fluidframework/tinylicious-client" / "@fluidframework/azure-client" synced to version 0.48.1 although latest ones are quite newer atm
Works with tinylicious as well as Azure Fluid Relay
Also started to [blog](https://mmsharepoint.wordpress.com/2022/05/31/fluidframework-in-a-collaborative-app-vote-movies/) about it...
